app/src/services/ShopService.ts

```diff
@@ -67,6 +71,21 @@ export const nextAvailableDate = (reservationDate: Date, menuDuration: number) =
   nextAvailableDate.setMinutes(nextAvailableDate.getMinutes() + menuDuration)
   return nextAvailableDate
 }
+export const checkAvailablity = (reservations: Reservation[], wantedReservationDate: Date) => {
+  if (reservations.length === 0) return [true]
+  const availability = reservations.map(reservations => {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every
you might wanna use this instead of map